akka-actor/src/main/scala/akka/pattern/StatusReply.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import akka.Done
+
+import scala.util.Try
+import scala.util.control.NoStackTrace
+import scala.util.{ Failure => ScalaFailure }
+import scala.util.{ Success => ScalaSuccess }
+
+/**
+ * Generic top-level message type for replies that signal failure or success. Convenient to use together with the
+ * `askWithStatus` ask variants.
+ *
+ * Create using the factory methods [[StatusReply#success]] and [[StatusReply#error]].
+ *
+ * Akka contains predefined serializers for the wrapper type and the textual error messages.
+ *
+ * @tparam T the type of value a successful reply would have
+ */
+final class StatusReply[+T] private (private val status: Try[T]) {
+
+  /**
+   * Java API: in the case of a successful reply returns the value, if the reply was not successful the exception
+   * the failure was created with is thrown
+   */
+  def getValue: T = status.get
+
+  /**
+   * Java API: returns the exception if the reply is a failure, or throws an exception if not.
+   */
+  def getError: Throwable = status match {
+    case ScalaFailure(ex) => ex
+    case _                => throw new IllegalArgumentException("Expected reply to be a failure, but was a success")
+  }
+
+  def isError: Boolean = status.isFailure
+  def isSuccess: Boolean = status.isSuccess
+
+  override def equals(other: Any): Boolean = other match {
+    case that: StatusReply[_] => status == that.status
+    case _                    => false
+  }
+
+  override def hashCode(): Int = status.hashCode
+
+  override def toString: String = status match {
+    case ScalaSuccess(t)  => s"Success($t)"
+    case ScalaFailure(ex) => s"Error(${ex.getMessage})"
+  }
+
+}
+
+object StatusReply {
+
+  /**
+   * Scala API: A general purpose message for using as an Ack
+   */
+  val Ack: StatusReply[Done] = success(Done)
```

Comment:
Yeah, that was my first thought but I think it will likely end up conflicting with `akka.Done` itself in code. You'd probably want to import this singleton rather than using the full `StatusReply.Ack` in actual code even if I used the object prefix in all samples to make them extra clear.

Reply:
Sure, let me try one more. 

What about `ReplyDone`? ;-)

Anyway, no blocker for PR.